Tests/SwiftSyntaxBuilderTest/StringInterpolationTests.swift

```diff
@@ -485,4 +486,40 @@ final class StringInterpolationTests: XCTestCase {
       )
     }
   }
+
+  func testInvalidSyntax() {
+    let invalid = DeclSyntax("return 1")
+    XCTAssert(invalid.hasError)
+
+    XCTAssertThrowsError(try DeclSyntax(validating: "return 1")) { error in
+      AssertStringsEqualWithDiff(
+        String(describing: error),
+        """
+
+        1 │ return 1
```

Comment:
I'd prefer not testing the formatted string here. That's going to be a pain if we ever change it 😅

Reply:
We already have tests that check the diagnostic format in `SwiftDiagnosticTests` and I really want to check that we produce a nice-looking error message here. 

So: I would-refer to keep it.